packages/devtools/lib/src/timeline/frames_bar_plotly.dart

```diff
@@ -0,0 +1,239 @@
+          dash: 'dash'),
+    );
+
+    Data high_water_mark_gpu = Data(
+      type: 'scatter',
+      x: [],
+      y: [],
+      hoverinfo: 'none',
+      name: 'GPU Jank',
+      showlegend: false,
+      line: Line(
+          color: GPU_COLOR,
+          width: 1,
+          dash: 'dash'),
+    );
+*/
+    // Return in trace index order
+    return [
+      trace_cpu_good,
+      trace_cpu_jank,
+      trace_gpu_good,
+      trace_gpu_jank,
+//      high_water_mark_cpu,
+//      high_water_mark_gpu
+    ];
+  }
+
+
+  void plotFPS() {
+    Plotly.newPlot(_domName, createFPSTraces(),
+        getFPSTimeseriesLayout('Flutter FPS Rate'), Configuration(responsive: true,
+            displaylogo: false,
+            modeBarButtonsToRemove: ['sendDataToCloud', 'select2d', 'lasso2d',
+            'hoverClosestCartesian', 'hoverCompareCartesian', 'toggleSpikelines']));
+  }
+
+  void plotFPSDataum(int dataIndex, num cpuDuration, num gpuDuration, bool paused) {
+    List<int> traces = [];
+
+    traces.add(cpuDuration > 8 ? CPU_JANK_TRACE : CPU_GOOD_TRACE);
+    traces.add(gpuDuration > 8 ? GPU_JANK_TRACE : GPU_GOOD_TRACE);
+//    traces.add(CPU_HIGH_WATER_MARK_TRACE);
+//    traces.add(GPU_HIGH_WATER_MARK_TRACE);
+
+    TraceData data = TraceData(
+      x: [[dataIndex], [dataIndex]],
+      y: [[cpuDuration], [gpuDuration]],
+//      x: [[dataIndex], [dataIndex], [dataIndex], [dataIndex]],
+//      y: [[cpuDuration], [gpuDuration], [8], [8]],
+    );
+
+    Plotly.extendTraces(_domName, data, traces);
+
+    if (!paused) rangeSliderToLast(dataIndex);
+  }
+
+  // Chunky ploting of data for reduce plotly live charting lag.
+  void plotFPSDataList(List<int> dataIndexes, List<num> cpuDurations, List<num> gpuDurations, bool paused) {
+    List<int> cpuGoodX = [];
+    List<num> cpuGoodTrace = [];
+
+    List<int> cpuJankX = [];
+    List<num> cpuJankTrace = [];
+
+    List<int> gpuGoodX = [];
+    List<num> gpuGoodTrace = [];
+
+    List<int> gpuJankX = [];
+    List<num> gpuJankTrace = [];
+
+    var totalIndexes = dataIndexes.length;
+    for (int dataIndex = 0; dataIndex < totalIndexes; dataIndex++)
+    {
+      num cpuDuration = cpuDurations[dataIndex];
+      num gpuDuration = gpuDurations[dataIndex];
+
+      if (cpuDuration > 8) {
+        cpuJankX.add(dataIndexes[dataIndex]);
+        cpuJankTrace.add(cpuDuration);
+      } else {
+        cpuGoodX.add(dataIndexes[dataIndex]);
+        cpuGoodTrace.add(cpuDuration);
+      }
+
+      if (gpuDuration > 8) {
+        gpuJankX.add(dataIndexes[dataIndex]);
+        gpuJankTrace.add(gpuDuration);
+      } else {
+        gpuGoodX.add(dataIndexes[dataIndex]);
+        gpuGoodTrace.add(gpuDuration);
+      }
+    };
+
+/*
+    TraceData data = TraceData(
+      x: [cpuJankX, cpuGoodX, gpuJankX, gpuGoodX],
+      y: [cpuJankTrace, cpuGoodTrace, gpuJankTrace, gpuGoodTrace],
+    );
+    Plotly.extendTraces(_domName, data, [CPU_JANK_TRACE, CPU_GOOD_TRACE, GPU_JANK_TRACE, GPU_GOOD_TRACE]);
+*/
```

Comment:
Can we remove all the commented out bits?

Reply:
Yep